examples/nlp_seq/models.py

```diff
@@ -90,146 +69,150 @@ def init(key, shape, dtype=np.float32):
-            attention_dropout_rate=0.1,
-            deterministic=False):
-    """Applies Transformer1DBlock module.
+  Args:
+    config: TransformerConfig dataclass containing hyperparameters.
+  """
+  config: TransformerConfig
+
+  @nn.compact
+  def __call__(self,
+               inputs,
+               encoder_mask=None,
+               deterministic=True):
+    """Applies Encoder1DBlock module.
 
     Args:
-      inputs: input data
-      qkv_dim: dimension of the query/key/value
-      mlp_dim: dimension of the mlp on top of attention block
-      num_heads: number of heads
-      causal_mask: bool, mask future or not
-      padding_mask: bool, mask padding tokens
-      dropout_rate: dropout rate
-      attention_dropout_rate: dropout rate for attention weights
-      deterministic: bool, deterministic or not (to apply dropout)
+      inputs: input data.
+      encoder_mask: encoder self-attention mask.
 
     Returns:
-      output after transformer block.
-
+      output after transformer encoder block.
     """
+    cfg = self.config
 
     # Attention block.
     assert inputs.ndim == 3
-    x = nn.LayerNorm(inputs)
+    x = nn.LayerNorm(dtype=cfg.dtype)(inputs)
     x = nn.SelfAttention(
-        x,
-        num_heads=num_heads,
-        qkv_features=qkv_dim,
-        attention_axis=(1,),
-        causal_mask=causal_mask,
-        padding_mask=padding_mask,
-        kernel_init=nn.initializers.xavier_uniform(),
-        bias_init=nn.initializers.normal(stddev=1e-6),
-        bias=False,
+        num_heads=cfg.num_heads,
+        dtype=cfg.dtype,
+        qkv_features=cfg.qkv_dim,
+        kernel_init=cfg.kernel_init,
+        bias_init=cfg.bias_init,
+        use_bias=False,
         broadcast_dropout=False,
-        dropout_rate=attention_dropout_rate,
-        deterministic=deterministic)
-    x = nn.dropout(x, rate=dropout_rate, deterministic=deterministic)
+        dropout_rate=cfg.attention_dropout_rate,
+        deterministic=deterministic)(x, encoder_mask)
+
+    x = nn.Dropout(rate=cfg.dropout_rate)(
+        x, deterministic=deterministic)
     x = x + inputs
 
     # MLP block.
-    y = nn.LayerNorm(x)
-    y = MlpBlock(
-        y,
-        mlp_dim=mlp_dim,
-        dropout_rate=dropout_rate,
-        deterministic=deterministic)
+    y = nn.LayerNorm(dtype=cfg.dtype)(x)
+    y = MlpBlock(config=cfg)(y, deterministic=deterministic)
 
     return x + y
 
 
 class Transformer(nn.Module):
   """Transformer Model for sequence tagging."""
 
-  def apply(self,
+  config: TransformerConfig
+
+  @nn.compact
+  def __call__(self,
             inputs,
-            vocab_size,
-            output_vocab_size,
-            emb_dim=512,
-            num_heads=8,
-            num_layers=6,
-            qkv_dim=512,
-            mlp_dim=2048,
-            max_len=2048,
-            train=True,
-            dropout_rate=0.3,
-            attention_dropout_rate=0.3):
+            train=False,
+            dropout_rate=0.3):
```

Comment:
Why is this not a setting in `TransformerConfig` ?

Reply:
now taken from TransformerConfig. Thanks!

Reply:
use dropout from config.